CONFIG.md

```diff
@@ -0,0 +1,151 @@
+# config
+
+The canonical documentation of shpool's config is the comments
+on the `Config` struct defined in `libshpool/src/config.rs`, but
+this document aims to provide some high level explanations of
+some common configuration options.
+
+You can specify the path to your config file by passing a
+`-c /path/to/config.toml` flag, or by creating and
+editing `~/.config/shpool/config.toml`.
+
+## Prompt Prefix
+
+By default, `shpool` will detect when you are using a shell it knows
+how to inject a prompt into. Currently, those shells include `bash`,
+`zsh` and `fish`, but more may be added in the future. If it noticed
+you are using one such shell, it will inject the prompt prefix
+`shpool:$SHPOOL_SESSION_NAME` at the beginning of your prompt
+in order to hint to you when you are inside of a `shpool` session.
+
+You can customize this prompt prefix by setting a new value in
+your config. For example, to show the `shpool` session name
+inside square brackets, you can put
+
+```
+prompt_prefix = "[$SHPOOL_SESSION_NAME]"
+```
+
+in your config file. If you want to instead completely suppress
+the prompt injection, you can just set a blank `prompt_prefix`
+with
+
+```
+prompt_prefix = ""
+```
+
+this allows you to write a custom prompt hook in your .rc files
+that examines the `$SHPOOL_SESSION_NAME` environment variable
```

Comment:
Usually a environment variable is not mentioned with the ` $`, the dollar is the way to access it, not it's name. You are not using `VAR=foo`, not `$VAR=foo` when defining it.

This remark is general and can be applied in all document